TYPO3.Neos/Classes/TYPO3/Neos/Domain/Repository/SiteRepository.php

```diff
@@ -54,4 +62,25 @@ public function findFirstOnline()
     {
         return $this->findOnline()->getFirst();
     }
+
+    /**
+     * Find the default site and fallback to first online
+     * if no default is found or the default is not online
+     */
+    public function findDefault()
+    {
+        if ($this->defaultSiteNodeName !== null) {
+            /**
+             * @var Site $defaultSite
+             */
+            $defaultSite = $this->findOneByNodeName($this->defaultSiteNodeName);
+            if ($defaultSite && $defaultSite->getState() === Site::STATE_ONLINE) {
+                return $defaultSite;
+            } else {
+                throw new NeosException(sprintf('DefaultSiteNode %s not found or not active', $this->defaultSiteNodeName), 1476374818);
+            }
+        } else {
```

Comment:
unnecessary else statement.. also we usually prefer exit early and avoid nesting..

e.g.

```php
if ($this->defaultSiteNodeName === null) {
    return $this->findOnline()->getFirst();
}

/** @var Site $defaultSite */
$defaultSite = $this->findOneByNodeName($this->defaultSiteNodeName);
if ($defaultSite && $defaultSite->getState() === Site::STATE_ONLINE) {
     return $defaultSite;
}
throw new NeosException(sprintf('DefaultSiteNode %s not found or not active', $this->defaultSiteNodeName), 1476374818);
```